source/Cosmos.Core/ACPI.cs

```diff
@@ -188,6 +263,44 @@ static bool Check_RSD(uint address)
 
             return sum == 0;
         }
+        /// <summary>
+        /// Parse the MADT
+        /// </summary>
+
+        private static void ParseMADT(byte* madtPtr)
+        {
+            MADT* madt = (MADT*)madtPtr;
+
+            byte* madtEnd = madtPtr + madt->Length;
+            byte* entryPtr = madtPtr + sizeof(MADT);
+
+            while (entryPtr < madtEnd)
+            {
+                MADTEntryHeader* entryHeader = (MADTEntryHeader*)entryPtr;
+
+                switch ((MADTEntryType)entryHeader->EntryType)
+                {
+                    case MADTEntryType.ProcessorLocalApic:
+                        ProcessorLocalApicEntry* localApic = (ProcessorLocalApicEntry*)entryPtr;
+                        if ((localApic->Flags & 1) != 0) // CPU is enabled
+                        {
+                            Console.WriteLine($"Found CPU with APIC ID: {localApic->ApicId}");
+                            // Here you can add this APIC ID to a list or process it as needed.
+                        }
+                        break;
+
+                    case MADTEntryType.IoApic:
+                        IoApicEntry* ioApic = (IoApicEntry*)entryPtr;
+                        Console.WriteLine($"Found IO APIC with ID: {ioApic->IoApicId} at address: {ioApic->IoApicAddress:X}");
+                        break;
+
+                        // Handle other MADT entries if needed
```

Comment:
You said "I MADE MADT WORK"
But these types of comments are usually from ChatGPT. Not judging, I use ChatGPT code too sometimes, but it's still not your code, stop pretending it is.

Reply:
No I Added the Comments because Last Time I comitted something they hated me because I removed all comments

Reply:
Only the Comments are Copilot Not the Code

Reply:
you didn't ask copilot to add comments, you asked it something like "add madt" and gave it the ACPI.cs. "// Handle other MADT entries if needed" does not come from an AI adding comments separately.